Realm.Shared/Dynamic/DynamicRealmObjectHelper.cs

```diff
@@ -25,6 +27,34 @@ internal class DynamicRealmObjectHelper : IRealmObjectHelper
     {
         internal static readonly DynamicRealmObjectHelper Instance = new DynamicRealmObjectHelper();
 
+        public void CopyToRealm(RealmObject instance)
```

Comment:
Objects whose helper is `DynamicRealmObjectHelper` cannot exist in an unmanaged state - they always belong to a realm. I think it's safe to make this method throw an exception instead.